### PR TITLE
E2E Tests: Fix tests in CI

### PR DESCRIPTION
**Pull request by brbrr**
Tests were failing mostly because of the new, unexpected _browser_ error. For some reason Chromium (only in headless mode?) thinks that Yahoo analytics script that we use on wordpress.com has some syntax error. Because of this error - browser was closing unexpectedly (since jest-puppeteer exits browser on error by default). 

#### Changes proposed in this Pull Request:

* Disable `exitOnPageError` for Puppeteer / Chromium
* Update some e2e packages
* add additional stuff to CI configuration step
* Update checkout page flow to use new iframed inputs

#### Testing instructions:

* You can try to run tests locally, just follow the instructions in Readme
* Check CI. it's green again! https://travis-ci.org/Automattic/jetpack/jobs/571903982

#### Proposed changelog entry for your changes:

* N/A

### CODE REVIEW

File: tests/e2e/jest-puppeteer.config.js [hunk 1]
```diff
@@ -4,15 +4,28 @@
...
+	dumpio = true;
+}
+
 module.exports = {
 	launch: {
 		headless: process.env.PUPPETEER_HEADLESS !== 'false',
 		devtools: process.env.PUPPETEER_HEADLESS === 'false',
 		slowMo: parseInt( process.env.PUPPETEER_SLOWMO, 10 ) || 0,
 		executablePath,
+		// args: [ '--no-sandbox' ],
```
Comment (oskosk) [inline tests/e2e/jest-puppeteer.config.js]: do we need this commented out ?

File: .travis.yml [hunk 2]
```diff
@@ -44,7 +44,9 @@ matrix:
     name: "E2E tests"
     if: branch !~ /(^branch-.*-built)/ AND env(RUN_E2E) = true
     before_script:
-    - nvm install
+    # Enable user namespace cloning
+    - "sysctl kernel.unprivileged_userns_clone=1"
+    - nvm install 10
```
Comment (simison) [inline .travis.yml]: Doesn't nvm file already define 10 also for Travis?

https://github.com/Automattic/jetpack/blob/master/.nvmrc

Comment (brbrr) [reply .travis.yml]: addressed in 476f303

File: tests/e2e/jest-puppeteer.config.js (hunk 1 above)
Comment (brbrr) [reply tests/e2e/jest-puppeteer.config.js]: addressed in 476f303

Comment (brbrr) [reply tests/e2e/jest-puppeteer.config.js]: addressed in 476f303

**Review by oskosk — APPROVED** [review]

**Pull request merged**